minoc/src/intTest/java/org/opencadc/minoc/IncorrectPutMetadataTest.java

```diff
@@ -304,7 +308,10 @@ public Object run() throws Exception {
                     // put file
                     InputStream in = new ByteArrayInputStream(bytes);
                     HttpUpload put = new HttpUpload(in, artifactURL);
-                    put.setRequestProperty(HttpTransfer.CONTENT_MD5, computeMD5(bytes));
+                    URI checksumURI = computeChecksumURI(bytes);
+                    String algorithm = checksumURI.getScheme();
+                    String checksum = DigestUtil.base64Encode(checksumURI.getSchemeSpecificPart());
+                    put.setRequestProperty(HttpTransfer.DIGEST, String.format("%s=%s", algorithm, checksum));
```

Comment:
Ack! `setDigest(URI)` ???

Reply:
HttpUpload has deprecated the individual set[Header] methods and uses setRequestProperty(name, value). But most header values are known and don't require computation, unlike the Digest header. Worthwhile to add a setDigest(URI checksum) to HttpUpload?

Reply:
This should use the `setDigest(URI)` method instead of re-implementing the http-specific encoding junk here. That method should be available for any method that sends body content (HttpPost and HttpPut)... whenever there is some code to receive something and provide it with a get method, there is probably a place for a symmetric set method to send such a value.